### DISCUSSION TITLE
Image resets to default settings

### DISCUSSION THREAD

**Discussion by JulianPrieber** [JulianPrieber/llc-docker#49] [Issues]
I've noticed that users instances reset to the default settings.

One explanation for this problem would be some system like Watchtower. **However, this does not appear the case, as this happens to users without any form of automatic updates.**

I'm not sure how to approach this or what could be causing this.

### If anyone can find a way to reliably reproduce this issue, please let me know.

**Comment by dnburgess** [comment]
As mentioned in our email conversation, I've had my instance reset and I'm not using WatchTower or any other update solution. Actually I hadn't even run an update on my system when I noticed the container had been reset. Looking forward to a solution when available. :)

**Comment by JulianPrieber** [comment]
I finally got around writing the guide for the persistent storage of the docker https://github.com/JulianPrieber/llc-docker/commit/0c72b576fa854ef9deebd2b78867c2f4ea751c41.

For me, this is a fairly simple thing that you can essentially do with one extra step. However, I have talked with many people about this and everyone seems to have a different opinion on how to do this.

 I tried to keep this explanation short while containing everything to know for people who care abut this to do it their way.

**If anyone wants to add anything feel free to open a pull request including anything I could have missed.**

**Comment by chrisjameschamp** [comment]
Hey @JulianPrieber I am also having this issue, although I cannot figure out how to add the persistent storage on an Unraid server, any help would be greatly appreciated

**Reply by JulianPrieber** [reply]
I could only help you're running Portainer on Unraid

**Reply by chrisjameschamp** [reply]
Hmm no I am not running portainer, ok I will do some more research

**Reply by chrisjameschamp** [reply]
are all files stored in "htdocs"? or just the customizable ones?

**Reply by JulianPrieber** [reply]
All application critical files are stored there.

**Comment by themitichris** [comment]
My solution (detailed) for Unraid:
_Permanent htdocs folder_

1. Create a folder in appdata named _littlelink_;
2. Open a Unraid terminal
`docker cp littlelink-custom:/htdocs /mnt/user/appdata/littlelink`
where _littlelink-custom_ is your container name (can be modified)
it will copy the entire folder in the new one created before.
3. Fix permissions (Laravel errors etc)
`chmod -R 777 /mnt/user/appdata/littlelink`
4. Go to edit from the menu of littlelink-custom container
5. Press on _+Add another Path, Port, Variable, Label or Device_ and enter
Name: `htdocs Path`
Container Path: `/htdocs`
Container Path: `/mnt/user/appdata/littlelink/htdocs`
6. Save

Done.

At this point you can safely update from GUI and stop/start containr without any emptying

**Reply by chrisjameschamp** [reply]
I did this, and after restarting the server I started getting 500 errors when trying to do anything

**Reply by themitichris** [reply]
> I did this, and after restarting the server I started getting 500 errors when trying to do anything

Can you show what you get?

**Comment by chrisjameschamp** [comment]
It stopped giving me the 500 errors for now, it appears to only do it if I
restore from a backup, but I would rather not do that at the moment since
everything is working fine for now

On Tue, Mar 21, 2023 at 1:28 PM Christopher Monti ***@***.***>
wrote:

> I did this, and after restarting the server I started getting 500 errors
> when trying to do anything
>
> Can you show what you get?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JulianPrieber/llc-docker/discussions/49#discussioncomment-5385694>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJIR2LOAEWSVE2ZGHHSZU3TW5IFOHANCNFSM6AAAAAAVQMBKJA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 

[image: logo]

CHRIS JAMES CHAMPEAU

Director / Editor / VFX Artist

(c) 920.737.0098

ChrisChampeau.com

***@***.***
|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||

This email, and any attachment, is intended only for the person or entity
to which it
is addressed and may contain confidential and/or privileged material. If
you have
received it in error please advise the sender immediately and then delete
it from
your system. The unauthorized use, distribution, copying or alteration of
this email
and any attachments is strictly prohibited. ChrisJamesChamp is a division
of Glass
Half Bourbon Inc. All creative works therein are subject to Copyright 2021
Glass
Half Bourbon Inc. All moral rights reserved.
